diff -Naur Gtk3-0.026/lib/Gtk3.pm Gtk3-0.026.tpg/lib/Gtk3.pm
--- Gtk3-0.026/lib/Gtk3.pm	2015-11-30 17:43:45.000000000 +0000
+++ Gtk3-0.026.tpg/lib/Gtk3.pm	2016-01-16 02:09:00.124172142 +0000
@@ -505,6 +505,17 @@
   Gtk3->export_to_level (1, @unknown_args);
 }
 
+# Gtk3 invokes gtk_main directly so Glib->enable_exceptions3 is not enough:
+sub enable_exceptions {
+    require Glib;
+    Glib->enable_exceptions3;
+    my $old_main = \&Gtk3::main;
+    *Gtk3::main = sub {
+       $old_main->();
+       Glib->generic_exceptions_handler();
+    };
+}
+
 # - Overrides --------------------------------------------------------------- #
 
 =item * For backwards compatibility, C<Gtk3::CHECK_VERSION>,
